File: Medium/2026-06-28-1846-Maximum-Element-After-Decreasing-and-Rearranging/solution.py
class Solution:
    def maximumElementAfterDecreasingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()
        max_elem = 1
        for num in arr:
            if num > max_elem + 1:
                num = max_elem + 1
            max_elem = num
        return max_elem

File: Medium/2026-06-28-1846-Maximum-Element-After-Decreasing-and-Rearranging/solution.java
class Solution {
    public int maximumElementAfterDecreasingAndRearranging(int[] arr) {
        Arrays.sort(arr);
        arr[0] = 1; // First element must be 1
        for (int i = 1; i < arr.length; i++) {
            arr[i] = Math.min(arr[i], arr[i - 1] + 1); // Ensure the difference condition
        }
        return arr[arr.length - 1]; // Return the maximum element
    }
}